fix(hashicorp/terraform-provider-google#23718): removing unnecessary …

sachinpro opened the pull request:
fixes hashicorp/terraform-provider-google#23718.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
bigquery: fixing crash during call to `expandParquetOptions` from redundant `resourceTable` call.
```

modular-magician commented:
#### Tests analytics
Total tests: 137
Passed tests: 125
Skipped tests: 10
Affected tests: 2

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>bigquery</li>

</ul>

</blockquote>
</details>
#### Action taken
<details>
<summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionIDAndHivePartitioning</li>
<li>TestAccBigQueryExternalDataTable_parquetOptions</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccBigQueryExternalDataTable_parquetOptions` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/3c42264b-6cc7-4d05-b025-53659f5061c1/recording/TestAccBigQueryExternalDataTable_parquetOptions.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---


🔴 Tests failed during RECORDING mode:
`TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionIDAndHivePartitioning` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/3c42264b-6cc7-4d05-b025-53659f5061c1/build-log/recording_build/TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionIDAndHivePartitioning_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/3c42264b-6cc7-4d05-b025-53659f5061c1/recording/TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionIDAndHivePartitioning.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/3c42264b-6cc7-4d05-b025-53659f5061c1/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/3c42264b-6cc7-4d05-b025-53659f5061c1/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/36d3be21710d0997967df0291cff2ab761714665..a6aca99cc4daddc5f93bacde80c0b04883ad414d) ( 2 files changed, 14 insertions(+), 4 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/7d4c100b9ed1fe46fc93d716d71ad30ddf33b5a7..253b118f79ae1cb97d5c5ce21e93cfd4c3e74562) ( 2 files changed, 14 insertions(+), 4 deletions(-))

modular-magician commented:
#### Tests analytics
Total tests: 137
Passed tests: 124
Skipped tests: 10
Affected tests: 3

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>bigquery</li>

</ul>

</blockquote>
</details>
#### Action taken
<details>
<summary>Found 3 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionIDAndHivePartitioning</li>
<li>TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionID_UpdateNoConnectionID</li>
<li>TestAccBigQueryExternalDataTable_CSV_WithSchema_UpdateAllowQuotedNewlines</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionID_UpdateNoConnectionID` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/c1ee9be5-b510-4291-9f7a-e48446bb3c72/recording/TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionID_UpdateNoConnectionID.log)]
`TestAccBigQueryExternalDataTable_CSV_WithSchema_UpdateAllowQuotedNewlines` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/c1ee9be5-b510-4291-9f7a-e48446bb3c72/recording/TestAccBigQueryExternalDataTable_CSV_WithSchema_UpdateAllowQuotedNewlines.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---


🔴 Tests failed during RECORDING mode:
`TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionIDAndHivePartitioning` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/c1ee9be5-b510-4291-9f7a-e48446bb3c72/build-log/recording_build/TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionIDAndHivePartitioning_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/c1ee9be5-b510-4291-9f7a-e48446bb3c72/recording/TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionIDAndHivePartitioning.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/c1ee9be5-b510-4291-9f7a-e48446bb3c72/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/c1ee9be5-b510-4291-9f7a-e48446bb3c72/recording) for each test

wj-chen commented:
From my understanding of the `TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionIDAndHivePartitioning` failure - Since we remove the generated columns on the TF side, when there is an Update, we only send the user-provided columns in the request, and that gets rejected by the server because it seems that the schema in the request is missing the generated columns.

One option for the fix may be when we filter out the generated columns in `Read()`, collect them and save separately into an output-only virtual field `generated_schema_columns` or so to be merged with `schema` for future Updates. Let me know what you think and if there are other ideas.

wj-chen commented:
Also add https://github.com/hashicorp/terraform-provider-google/issues/23655 to "fixes" in Description and update the release note (e.g. "fixing" -> "fixed").

sachinpro commented:
> Also add [hashicorp/terraform-provider-google#23655](https://github.com/hashicorp/terraform-provider-google/issues/23655) to "fixes" in Description and update the release note (e.g. "fixing" -> "fixed").

Done.

sachinpro commented:
Note that the first [commit](https://github.com/GoogleCloudPlatform/magic-modules/pull/14659/commits/1806bd356f2aa9f6b3b279583475f4e7a367104b) fixes both the issues I mentioned in the description. The other commit is only for fixing a test which was already failing.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/34218f1f0b2c3018ba1c6c3ca6f44efa59ee29b3..a402300d970c9108b561aff4f9270e3773cad1fe) ( 2 files changed, 154 insertions(+), 98 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/a1cdf7838d4e396d0cda27665d967f8a2c2dfcc0..9adf40a8e033bcef9400c3abc44cc8da6a8c804f) ( 2 files changed, 154 insertions(+), 98 deletions(-))





## Missing doc report (experimental)


The following resources have fields missing in documents. 

- `google_bigquery_table` 
  - Expected Document Path: `/website/docs/r/bigquery_table.html.markdown`
  - Fields: `[generated_schema_columns]`

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionIDAndHivePartitioning` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/bf9421f0-abc7-43b2-979c-7471907b1028/recording/TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionIDAndHivePartitioning.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---






🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/bf9421f0-abc7-43b2-979c-7471907b1028/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/bf9421f0-abc7-43b2-979c-7471907b1028/recording) for each test

sachinpro commented:
> > > From my understanding of the `TestAccBigQueryExternalDataTable_CSV_WithSchemaAndConnectionIDAndHivePartitioning` failure - Since we remove the generated columns on the TF side, when there is an Update, we only send the user-provided columns in the request, and that gets rejected by the server because it seems that the schema in the request is missing the generated columns.
> > > One option for the fix may be when we filter out the generated columns in `Read()`, collect them and save separately into an output-only virtual field `generated_schema_columns` or so to be merged with `schema` for future Updates. Let me know what you think and if there are other ideas.
> > 
> > 
> > @wj-chen I have added this logic in the [commit](https://github.com/GoogleCloudPlatform/magic-modules/pull/14659/commits/ab3d860c3ba682d8051693f89c79e384d8163e38).
> > It is unfortunate that we need to ignore the virtual fields in all our tests.
> 
> Thank you for fixing this!
> 
> To keep documentations consistent, let's add `ignore_auto_generated_schema` and `generated_schema_columns` to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/bigquery/Table.yaml, and `generated_schema_columns` to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/website/docs/r/bigquery_table.html.markdown?plain=1. Otherwise the change LGTM.

I have added them to virtual fields. 

As we discussed, to avoid confusion, I am not updating the documentation to include `generated_schema_columns`.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/2a823c092e99e9abcc10eedc2729d501a8a95ba9..2b00256fec7246d4b53a15f027694f0d2862eec5) ( 2 files changed, 154 insertions(+), 98 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/0da7d182981e1131eaee2a1f4156257e11b58d30..36b273ff4a34e03800973c6cd83d0e542799f29d) ( 2 files changed, 154 insertions(+), 98 deletions(-))





## Missing doc report (experimental)


The following resources have fields missing in documents. 

- `google_bigquery_table` 
  - Expected Document Path: `/website/docs/r/bigquery_table.html.markdown`
  - Fields: `[generated_schema_columns]`

modular-magician commented:
#### Tests analytics
Total tests: 137
Passed tests: 127
Skipped tests: 10
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>bigquery</li>

</ul>

</blockquote>
</details>
🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14659/artifacts/7a8fe637-dcf2-4639-b55f-0dd402ee66ac/build-log/replaying_test.log)
